### ISSUE TITLE
Constraining investment amount in a category to 0 results in "incompatible inequality constraints" error

### ISSUE THREAD

**Issue by tapajyoti-ghosh** [NREL/tyche#178]
For the constraints on the max amount invested per category we cannot have them as 0 but very low numbers. For example - 

```
w = optimizer.opt_slsqp(
    "LCOE",
    sense='min',
    total_amount = 5000000,
    max_amount = pd.Series(
        [2000000, 0, 5000000],
        index=evaluator.max_amount.index.tolist()
))
print(w[1])
print(w[2])
print(w[3])

```

This does not work.

**Comment by tapajyoti-ghosh** [comment]
But this works

```
w = optimizer.opt_slsqp(
    "LCOE",
    sense='min',
    total_amount = 5000000,
    max_amount = pd.Series(
        [2000000, 0.0000001, 5000000],
        index=evaluator.max_amount.index.tolist()
))
print(w[1])
print(w[2])
print(w[3])

```


This does not work.

**Comment by rjhanes** [comment]
Sure, I can look into this.

I'm assuming "does not work" means that the optimization is infeasible? If not and there's an error code/warning, please provide. 

Is this bug observed when any research area amount is set to zero, and does it happen consistently across objectives and across decision contexts?

**Comment by tapajyoti-ghosh** [comment]
<img width="1122" alt="image" src="https://github.com/NREL/tyche/assets/24482959/bf4fea80-871a-40f8-91e2-6ba6e3645ce5">

This one fails with the errors message.

**Comment by tapajyoti-ghosh** [comment]
Same one with no error message - 
<img width="1037" alt="image" src="https://github.com/NREL/tyche/assets/24482959/afd727a7-c030-4813-8ee1-1f445b1058b8">

**Comment by rjhanes** [comment]
Testing first on pv-residential-simple where the above bug was observed.

Best guess as to what's happening: Setting an investment amount to zero creates an equality constraint, and the optimization problem as coded in `EpsilonConstraintOptimizer` doesn't explicity include equality constraints.

Investment amount constraints are used to create bounds on decision variables. According to the ([docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.Bounds.html)), `Bounds` object can be used to define equality constraints, but that object is not currently being used. `bounds` the parameter is instead a plain list of tuples.

~~Fix attempt 1: Import the `Bounds` object and use it to create the `bounds` parameter.~~ Failed because `EpsilonConstraintOptimizer` uses `fmin_slsqp` (or whatever algorithm` rather than `scipy.Optimize.minimize`. So `bounds` has to be "old-style" rather than a `Bounds` object.

Pre-fix attempt 2: What's the mathematical and practical difference between optima with investment amounts constrained to 0 and to 1? Can we implement some back-end trickery to replace 0 with 1 without changing the resulting solution? Want to try this because IRL funding assigned wouldn't be $1 -there'd be no practical difference between zero funding and $1 funding. 

The replace-zero-with-non-zero-small-number worked: once the investment amount constraint was set below 1, the optimal amount in the solution became 0 even if that category had been invested in without the constraint.

Pushing a quick fix to `dev` to resolve this.

Note for future: The code might get more compact and easier to work on if we switch from using the `fmin_solver` function calls to the `scipy.Optimize` class. This would let us re-use parts of the problem between solvers that right now is accomplished with identical code lines, and would also let us use the `Bounds` object to define equality constraints, making this fix unnecessary. I think it would be too much work for right now, though it may become necessary for a future `scipy` release.

**Comment by rjhanes** [comment]
Closed in [d4553f3](https://github.com/NREL/tyche/commit/d4553f39a6349a31e7ce3a5507ec93ff7f9d3e8a)
